src/pcrlock/pcrlock.c

```diff
@@ -4974,6 +4986,10 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_pager_flags |= PAGER_DISABLE;
                         break;
 
+                case 'f':
+                        abbrev_hash = 0;
```

Comment:
I think conceptually it makes more sense to set this to SIZE_MAX to indicate that no abbreviation should take place.

Coincidentally, if you adjust it like that #31651 will make sure you can just pass it to strshorten() and not even think about it anymore.